scrapper.py

```diff
@@ -25,39 +40,154 @@ class Crawler:
     """
     Crawler implementation
     """
+
     def __init__(self, seed_urls, max_articles: int):
-        pass
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        """
+        get link to the article
+        """
+        for article_link in article_bs.find_all("a", class_="article__title"):
+            self.urls.append(DOMAIN + article_link["href"])
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
 
-    def get_search_urls(self):
-        """
-        Returns seed_urls param
-        """
-        pass
+        for seed_url in self.seed_urls:
+            response = requests.get(seed_url, headers=HEADERS)
+            sleep_period = random.randrange(3, 7)
+            time.sleep(sleep_period)
+
+            if not response.ok:
+                print("Request was unsuccessful.")
+                continue
+
+            seed_bs = BeautifulSoup(response.text, features="html.parser")
+            self._extract_url(seed_bs)
+
+
+def get_search_urls(self):
+    """
+    Returns seed_urls param
+    """
+    return self.seed_urls
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+
+    path = pathlib.Path(base_path)
+
+    if path.exists():
+        shutil.rmtree(path)
+
+    path.mkdir(parents=True)
 
 
 def validate_config(crawler_path):
     """
-    Validates given config
+    Validates config
     """
-    pass
+
+    with open(crawler_path) as file:
+        config = json.load(file)
+
+    urls = config["seed_urls"]
+    articles = config["total_articles_to_find_and_parse"]
+    http_regex = r"http[s]?://journals\.kantiana\.ru/."
+
+    if not urls:
+        raise IncorrectURLError
+
+    if not isinstance(articles, int) or articles <= 0:
+        raise IncorrectNumberOfArticlesError
+
+    if articles > 100:
+        raise NumberOfArticlesOutOfRangeError
+
+    for url in urls:
+        check = re.search(http_regex, url)
+        if not check:
+            raise IncorrectURLError
+
+    prepare_environment(ASSETS_PATH)
```

Comment:
follow the single responsibility principle: a function that *validates* should not *prepare* environment